radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/rx/ModuleRunnerImpl.java

```diff
@@ -102,13 +104,29 @@ private ModuleRunnerImpl(
 			return this;
 		}
 
-		public Builder onStart(Consumer<ScheduledExecutorService> r) {
-			this.onStart = r;
+		public <T> Builder scheduleWithFixedDelay(
+			EventDispatcher<T> eventDispatcher,
+			Supplier<T> eventSupplier,
+			Duration initialDelay,
+			Duration interval
+		) {
+			return onStart(executor ->
+				executor.scheduleWithFixedDelay(
+					() -> eventDispatcher.dispatch(eventSupplier.get()),
+					initialDelay.toMillis(),
+					interval.toMillis(),
+					TimeUnit.MILLISECONDS
+				)
+			);
+		}
+
+		public Builder onStart(Consumer<ScheduledExecutorService> fn) {
```

Comment:
Weird interface here, executorService should not be visible to builder interface but I suppose this is good enough for now.